### ISSUE TITLE
Sort Goby Hash Keys in Certain Order

### ISSUE THREAD

**Issue by Maxwell-Alexius** [goby-lang/goby#265]
The hash pair order in REPL seems not corresponding to the input order, e.g., when I try to print out simple hash:

```Ruby
$ go run goby.go -i
Goby 0.0.9
» h = {a: "Hello", b: "World", c: "Goby"}
#=> <Instance of: Object>
» h
#=> { c: Goby, b: World, a: Hello }
» h
#=> { c: Goby, b: World, a: Hello }
» h
#=> { a: Hello, c: Goby, b: World }
» h
#=> { b: World, a: Hello, c: Goby }
» h
#=> { b: World, a: Hello, c: Goby }
» h
#=> { b: World, a: Hello, c: Goby }
» h
#=> { c: Goby, b: World, a: Hello }
» h
#=> { c: Goby, b: World, a: Hello }
»  
```

Any idea?

**Comment by st0012** [comment]
@Maxwell-Alexius This is because we use Go's map to store pairs, and in Go map's key/value are random ordered.

**Comment by Maxwell-Alexius** [comment]
I'll try to find a solution for this case, thanks!

**Comment by Maxwell-Alexius** [comment]
@st0012 Maybe we can sort the keys in alphabetical order so that we can guarantee that the hash will be appear in certain order in Goby lang?

https://stackoverflow.com/questions/23330781/sort-golang-map-values-by-keys

**Comment by st0012** [comment]
@Maxwell-Alexius I think we should just sort them in `toString` method, and keep it random in other places. I remember I read a post about why Go made this random decision but I can't find it now. @hachi8833 Do you know why?

**Comment by hachi8833** [comment]
Looks like [the article](https://nathanleclaire.com/blog/2014/04/27/a-surprising-feature-of-golang-that-colored-me-impressed/) is the one.

* As for Ruby, the order of Hash items had not been guaranteed by 1.8: [1.9 or higher](https://www.igvita.com/2009/02/04/ruby-19-internals-ordered-hash/) guarantees the order of Hash items.

**Comment by Maxwell-Alexius** [comment]
@st0012 @hachi8833 Thanks, I think I can adjust this feature request which is listed below:

- [ ] Proposal I. There is no need to sort or order the Hash key-value pairs. However, for some method implementation, like `Hash#keys`, we can return an array of keys of hash in alphabetical order. (Affect some Hash API method implementation in #247)

- [ ] Proposal II. Remain current arbitrary state of the key-value pairs in hash API, so if implementing `Hash#keys`, the keys may also in an arbitrary order. (Issue closed here, if anyone wants to change the Hash pairs order, they can reopen this issue)

**Comment by st0012** [comment]
I actually agree that the hash shouldn't guarantee the key's order, so I suggest keep it this way. But maybe we can provide some extra API like `sorted_keys` for those who need the keys sorted.

**Comment by Maxwell-Alexius** [comment]
Okay, got it ~

**Comment by Maxwell-Alexius** [comment]
This should be closed because PR #247 completed.
